sound/soc/sof/sof-audio.c

```diff
@@ -301,6 +302,24 @@ static int sof_setup_pipeline_connections(struct snd_sof_dev *sdev,
 		}
 	}
 
+	/*
+	 * During route setup above, playback stream routes are set up by going
+	 * through sink path, capture stream routes are set up by going through
+	 * source path. Obviously, feedback route from a widget in the capture
+	 * stream to another widget in the playback stream is not able to be setup.
+	 *
+	 * The feedback route is setup here if widgets on both sides of the route
+	 * are active.
+	 */
+	list_for_each_entry(sroute, &sdev->route_list, list)
+		if (sroute->src_widget->use_count && sroute->sink_widget->use_count &&
+		    !sroute->setup) {
```

Comment:
this sounds extremely racy. What makes you think the use_count is safe to test like this? the source and sink widget use counts could be modified at any time.